### ISSUE TITLE
AM and PM are reverse when displayed in Lovelace from an integration - temperature

### ISSUE THREAD

**Issue by guitarflipper** [home-assistant/home-assistant.github.io#18954]
### Feedback

![image](https://user-images.githubusercontent.com/16021251/129595260-aaacfa4b-4651-4207-9bac-d6ac0207d522.png)

You can see from this screen capture that when you click on the weather card in lovelace that the AM temperature and PM temperatures are reversed. The integration is National Weather Service. I used KLXT weather station from a local airport about 2 miles away.

### URL

https://www.home-assistant.io/integrations/nws/

### Version

2021.8.7

**Comment by MatthewFlamm** [comment]
This is the more-info display after you click on a weather card?

As the documentation states, the time stamps are the times at which the forecast starts.  So these are not flipped.  However, it would be better to display Day vs Night like in the weather card, this is more of frontend feature request in my opinion.

**Comment by guitarflipper** [comment]
Yes, when the weather card is clicked. Can you point me to the documentation you mention? I don't see that on the NWS integration documentation page. I really don't understand your explanation. Sorry.

**Comment by MatthewFlamm** [comment]
> Can you point me to the documentation you mention? I don't see that on the NWS integration documentation page. I really don't understand your explanation. Sorry.

You linked to the documentation yourself.  "The time supplied for each forecast is the start time for the forecast.". This means the forecast with time stamp of 6 am is the forecast from 6 am to 6 pm.  Thus the high temperature with time stamp of 6 am is correct.

**Comment by guitarflipper** [comment]
So, if I understand correctly, when referring to the "time supplied", you mean the highest temperature for that period starting at 6AM. That's too bad because intuitively, one would look at the display and assume the opposite. I think the next course of action is for a feature request, as you stated. Thanks for helping me understand this better. "You are a gentleman and a scholar" (a compliment from my early years).
